src/app/forms/validators/custom-async-validators.spec.ts

```diff
@@ -114,5 +115,103 @@ describe('updateTestStationDetails', () => {
     expect(form.controls['testStationType'].value).toBe('bar')
     expect(form.controls['testStationName'].value).toBe('foo')
   });
-});
+}
+
+);
+
+describe('requiredIfNotFail', () => {
+  let form: FormGroup;
+  let store: MockStore<State>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [provideMockStore({ initialState: initialAppState })]
+    });
+
+    store = TestBed.inject(MockStore);
+
+    form = new FormGroup({
+      foo: new CustomFormControl({ name: 'foo', type: FormNodeTypes.CONTROL, children: [] }, null)
+    });
+  });
+
+  it('should be required when value is "pass"', async () => {
+    form.controls['foo'].patchValue('');
+
+    const testResult = mockTestResult();
```

Comment:
if all you need some the `testResult` here is the result of the test i.e.: pass, fail etc..., you can do 
```js
const testResult = {testTypes: [{testResult: resultOfTestEnum.pass}]} as TestResultModel 
```
instead of passing the whole `mockTestResult()` everytime. And if you need only 2-3 more properties, you should be able to pass them in the same fashion.

If you need much more than 2-3 properties in the test, you could consider assigning a variable in the `beforeEach` just as 
```js
let mockTestResult = mockTestResult()
```
and then change each testResult to the resultOfTest you need in each test

Reply:
Thanks that change has been made.